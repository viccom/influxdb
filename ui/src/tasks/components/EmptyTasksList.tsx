// Libraries
import React, {PureComponent} from 'react'

// Components
import {
  Button,
  ComponentColor,
  ComponentSize,
  EmptyState,
  IconFont,
} from 'src/clockface'

interface Props {
  searchTerm: string
  onCreate: () => void
  totalCount: number
}

export default class EmptyTasksLists extends PureComponent<Props> {
  public render() {
    const {searchTerm, onCreate, totalCount} = this.props

    if (totalCount && searchTerm === '') {
      return (
        <EmptyState size={ComponentSize.Large}>
          <EmptyState.Text
            text={`All ${totalCount} of your Tasks are inactive`}
          />
        </EmptyState>
      )
    }

    if (searchTerm === '') {
      return (
        <EmptyState size={ComponentSize.Large}>
          <EmptyState.Text
            text={"Looks like you don't have any Tasks , why not create one?"}
            highlightWords={['Tasks']}
          />
          <Button
            size={ComponentSize.Medium}
            color={ComponentColor.Primary}
            icon={IconFont.Plus}
            text="Create Task"
            onClick={onCreate}
          />
        </EmptyState>
      )
    }

    return (
      <EmptyState size={ComponentSize.Large}>
        <EmptyState.Text text={'No Tasks match your search term'} />
      </EmptyState>
    )
  }
}
