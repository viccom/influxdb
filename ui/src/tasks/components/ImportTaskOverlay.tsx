import React, {PureComponent} from 'react'
import _ from 'lodash'

import Container from 'src/clockface/components/overlays/OverlayContainer'
import Heading from 'src/clockface/components/overlays/OverlayHeading'
import Body from 'src/clockface/components/overlays/OverlayBody'
import DragAndDrop from 'src/shared/components/DragAndDrop'

interface Props {
  onDismissOverlay: () => void
  onSave: (script: string, fileName: string) => void
}

class ImportTaskOverlay extends PureComponent<Props> {
  constructor(props: Props) {
    super(props)
  }

  public render() {
    const {onDismissOverlay} = this.props

    return (
      <Container maxWidth={800}>
        <Heading title="Import Task" onDismiss={onDismissOverlay} />
        <Body>
          <DragAndDrop
            submitText="Upload Task"
            fileTypesToAccept={this.validFileExtension}
            handleSubmit={this.handleUploadTask}
          />
        </Body>
      </Container>
    )
  }

  private get validFileExtension(): string {
    return '.flux'
  }

  private handleUploadTask = (
    uploadContent: string,
    fileName: string
  ): void => {
    const {onSave, onDismissOverlay} = this.props
    onSave(uploadContent, fileName)
    onDismissOverlay()
  }
}

export default ImportTaskOverlay
