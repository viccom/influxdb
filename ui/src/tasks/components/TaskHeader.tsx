import React, {PureComponent} from 'react'

import {Page} from 'src/pageLayout'
import {ComponentColor, Button, ComponentStatus} from 'src/clockface'

import 'src/tasks/components/TasksPage.scss'

interface Props {
  title: string
  canSubmit: boolean
  onCancel: () => void
  onSave: () => void
}

export default class TaskHeader extends PureComponent<Props> {
  public render() {
    return (
      <Page.Header fullWidth={true}>
        <Page.Header.Left>
          <Page.Title title={this.props.title} />
        </Page.Header.Left>
        <Page.Header.Right>
          <Button
            color={ComponentColor.Default}
            text="Cancel"
            onClick={this.props.onCancel}
          />
          <Button
            color={ComponentColor.Success}
            text="Save"
            status={
              this.props.canSubmit
                ? ComponentStatus.Default
                : ComponentStatus.Disabled
            }
            onClick={this.props.onSave}
          />
        </Page.Header.Right>
      </Page.Header>
    )
  }
}
