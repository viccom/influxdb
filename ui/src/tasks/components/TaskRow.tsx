// Libraries
import React, {PureComponent} from 'react'
import {withRouter, WithRouterProps} from 'react-router'

// Components
import {
  ComponentSpacer,
  Alignment,
  Button,
  ComponentSize,
  SlideToggle,
  IndexList,
  ConfirmationButton,
  Stack,
  Label,
} from 'src/clockface'

// Utils
import {downloadTextFile} from 'src/shared/utils/download'
import {Task as TaskAPI, Organization} from 'src/api'

interface Task extends TaskAPI {
  organization: Organization
}

// Constants
import {IconFont} from 'src/clockface/types/index'

interface Props {
  task: Task
  onActivate: (task: Task) => void
  onDelete: (task: Task) => void
  onSelect: (task: Task) => void
  onEditLabels: (task: Task) => void
}

export class TaskRow extends PureComponent<Props & WithRouterProps> {
  public render() {
    const {task, onDelete} = this.props

    return (
      <IndexList.Row disabled={!this.isTaskActive}>
        <IndexList.Cell>
          <ComponentSpacer stackChildren={Stack.Rows} align={Alignment.Left}>
            <a href="#" onClick={this.handleClick}>
              {task.name}
            </a>
            {this.labels}
          </ComponentSpacer>
        </IndexList.Cell>
        <IndexList.Cell>
          <SlideToggle
            active={this.isTaskActive}
            size={ComponentSize.ExtraSmall}
            onChange={this.changeToggle}
          />
        </IndexList.Cell>
        <IndexList.Cell>{this.schedule}</IndexList.Cell>
        <IndexList.Cell>
          <a href="" onClick={this.handleOrgClick}>
            {task.organization.name}
          </a>
        </IndexList.Cell>
        <IndexList.Cell alignment={Alignment.Right} revealOnHover={true}>
          <ComponentSpacer align={Alignment.Right}>
            <Button
              size={ComponentSize.ExtraSmall}
              text="Export"
              icon={IconFont.Export}
              onClick={this.handleExport}
            />
            <ConfirmationButton
              size={ComponentSize.ExtraSmall}
              text="Delete"
              confirmText="Confirm"
              onConfirm={onDelete}
              returnValue={task}
            />
          </ComponentSpacer>
        </IndexList.Cell>
      </IndexList.Row>
    )
  }

  private handleClick = e => {
    e.preventDefault()

    this.props.onSelect(this.props.task)
  }

  private handleExport = () => {
    const {task} = this.props
    downloadTextFile(task.flux, `${task.name}.flux`)
  }

  private handleOrgClick = () => {
    const {router, task} = this.props
    router.push(`/organizations/${task.organization.id}/members_tab`)
  }

  private get labels(): JSX.Element {
    const {task} = this.props

    if (!task.labels.length) {
      return (
        <Label.Container
          limitChildCount={4}
          className="index-list--labels"
          onEdit={this.handleEditLabels}
        />
      )
    }

    return (
      <Label.Container
        limitChildCount={4}
        resourceName="this Task"
        onEdit={this.handleEditLabels}
      >
        {task.labels.map(label => (
          <Label
            key={label.name}
            id={label.resourceID}
            colorHex={label.properties.color}
            name={label.name}
            description={label.properties.description}
          />
        ))}
      </Label.Container>
    )
  }

  private get isTaskActive(): boolean {
    const {task} = this.props
    if (task.status === TaskAPI.StatusEnum.Active) {
      return true
    }
    return false
  }

  private handleEditLabels = () => {
    const {task, onEditLabels} = this.props

    onEditLabels(task)
  }

  private changeToggle = () => {
    const {task, onActivate} = this.props
    if (task.status === TaskAPI.StatusEnum.Active) {
      task.status = TaskAPI.StatusEnum.Inactive
    } else {
      task.status = TaskAPI.StatusEnum.Active
    }
    onActivate(task)
  }

  private get schedule(): string {
    const {task} = this.props
    if (task.every && task.offset) {
      return `Every ${task.every}, Offset ${task.offset}`
    }
    if (task.every) {
      return `Every ${task.every}`
    }
    if (task.cron) {
      return task.cron
    }
    return ''
  }
}
export default withRouter(TaskRow)
