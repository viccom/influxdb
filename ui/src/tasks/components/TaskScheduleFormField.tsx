// Libraries
import React, {PureComponent, ChangeEvent} from 'react'

// Components
import {Input, InputType, Grid, Form, Columns} from 'src/clockface'

// Types
import {TaskSchedule} from 'src/utils/taskOptionsToFluxScript'

interface Props {
  schedule: TaskSchedule
  cron: string
  offset: string
  interval: string
  onChangeInput: (e: ChangeEvent<HTMLInputElement>) => void
}

export default class TaskScheduleFormFields extends PureComponent<Props> {
  public render() {
    const {offset, onChangeInput, interval, cron, schedule} = this.props

    return (
      <>
        <Grid.Column widthXS={Columns.Six}>
          <Form.Element
            label={schedule === TaskSchedule.interval ? 'Interval' : 'Cron'}
          >
            <Input
              name={schedule}
              type={InputType.Text}
              placeholder={
                schedule === TaskSchedule.interval ? '1d3h30s' : '0 2 * * *'
              }
              value={schedule === TaskSchedule.interval ? interval : cron}
              onChange={this.props.onChangeInput}
            />
          </Form.Element>
        </Grid.Column>

        <Grid.Column widthXS={Columns.Six}>
          <Form.Element label="Offset">
            <Input
              name="offset"
              type={InputType.Text}
              value={offset}
              placeholder="20m"
              onChange={onChangeInput}
            />
          </Form.Element>
        </Grid.Column>
      </>
    )
  }
}
