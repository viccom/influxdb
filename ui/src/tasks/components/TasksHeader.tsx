// Libraries
import React, {PureComponent} from 'react'
import {Page} from 'src/pageLayout'

// Components
import {
  Button,
  ComponentColor,
  IconFont,
  ComponentSize,
  SlideToggle,
} from 'src/clockface'
import SearchWidget from 'src/shared/components/search_widget/SearchWidget'
import TaskOrgDropdown from 'src/tasks/components/TasksOrgDropdown'

import 'src/tasks/components/TasksPage.scss'

interface Props {
  onCreateTask: () => void
  setSearchTerm: (searchTerm: string) => void
  setShowInactive: () => void
  showInactive: boolean
  toggleOverlay: () => void
}

export default class TasksHeader extends PureComponent<Props> {
  public render() {
    const {
      onCreateTask,
      setSearchTerm,
      setShowInactive,
      showInactive,
      toggleOverlay,
    } = this.props

    return (
      <Page.Header fullWidth={false}>
        <Page.Header.Left>
          <Page.Title title="Tasks" />
        </Page.Header.Left>
        <Page.Header.Right>
          <SlideToggle.Label text="Show Inactive" />
          <SlideToggle
            active={showInactive}
            size={ComponentSize.ExtraSmall}
            onChange={setShowInactive}
          />
          <SearchWidget
            placeholderText="Filter tasks by name..."
            onSearch={setSearchTerm}
          />
          <TaskOrgDropdown />
          <Button
            text="Import"
            icon={IconFont.Import}
            onClick={toggleOverlay}
          />
          <Button
            color={ComponentColor.Primary}
            onClick={onCreateTask}
            icon={IconFont.Plus}
            text="Create Task"
            titleText="Create a new Task"
          />
        </Page.Header.Right>
      </Page.Header>
    )
  }
}
