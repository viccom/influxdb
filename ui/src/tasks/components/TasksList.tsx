// Libraries
import React, {PureComponent} from 'react'
import _ from 'lodash'

// Components
import {IndexList} from 'src/clockface'
import TaskRow from 'src/tasks/components/TaskRow'
import SortingHat from 'src/shared/components/sorting_hat/SortingHat'
import {OverlayTechnology} from 'src/clockface'
import EditLabelsOverlay from 'src/shared/components/EditLabelsOverlay'

// Types
import EmptyTasksList from 'src/tasks/components/EmptyTasksList'
import {Task as TaskAPI, User, Organization} from 'src/api'

interface Task extends TaskAPI {
  organization: Organization
  owner?: User
}
import {Sort} from 'src/clockface'
import {addTaskLabelsAsync, removeTaskLabelsAsync} from 'src/tasks/actions/v2'

interface Props {
  tasks: Task[]
  searchTerm: string
  onActivate: (task: Task) => void
  onDelete: (task: Task) => void
  onCreate: () => void
  onSelect: (task: Task) => void
  totalCount: number
  onRemoveTaskLabels: typeof removeTaskLabelsAsync
  onAddTaskLabels: typeof addTaskLabelsAsync
}

type SortKey = keyof Task | 'organization.name'

interface State {
  sortKey: SortKey
  sortDirection: Sort
  taskLabelsEdit: Task
  isEditingTaskLabels: boolean
}

export default class TasksList extends PureComponent<Props, State> {
  constructor(props) {
    super(props)
    this.state = {
      sortKey: null,
      sortDirection: Sort.Descending,
      taskLabelsEdit: null,
      isEditingTaskLabels: false,
    }
  }

  public render() {
    const {searchTerm, onCreate, totalCount} = this.props
    const {sortKey, sortDirection} = this.state

    const headerKeys: SortKey[] = [
      'name',
      'status',
      'every',
      'organization.name',
    ]

    return (
      <>
        <IndexList>
          <IndexList.Header>
            <IndexList.HeaderCell
              columnName="Name"
              width="20%"
              sortKey={headerKeys[0]}
              sort={sortKey === headerKeys[0] ? sortDirection : Sort.None}
              onClick={this.handleClickColumn}
            />
            <IndexList.HeaderCell
              columnName="Active"
              width="10%"
              sortKey={headerKeys[1]}
              sort={sortKey === headerKeys[1] ? sortDirection : Sort.None}
              onClick={this.handleClickColumn}
            />
            <IndexList.HeaderCell
              columnName="Schedule"
              width="20%"
              sortKey={headerKeys[2]}
              sort={sortKey === headerKeys[2] ? sortDirection : Sort.None}
              onClick={this.handleClickColumn}
            />
            <IndexList.HeaderCell
              columnName="Owner"
              width="15%"
              sortKey={headerKeys[3]}
              sort={sortKey === headerKeys[3] ? sortDirection : Sort.None}
              onClick={this.handleClickColumn}
            />
            <IndexList.HeaderCell columnName="" width="35%" />
          </IndexList.Header>
          <IndexList.Body
            emptyState={
              <EmptyTasksList
                searchTerm={searchTerm}
                onCreate={onCreate}
                totalCount={totalCount}
              />
            }
            columnCount={5}
          >
            {this.sortedRows}
          </IndexList.Body>
        </IndexList>
        {this.renderLabelEditorOverlay}
      </>
    )
  }

  private handleClickColumn = (nextSort: Sort, sortKey: SortKey) => {
    this.setState({sortKey, sortDirection: nextSort})
  }

  private rows = (tasks: Task[]): JSX.Element => {
    const {onActivate, onDelete, onSelect} = this.props
    const taskrows = (
      <>
        {tasks.map(t => (
          <TaskRow
            key={`task-id--${t.id}`}
            task={t}
            onActivate={onActivate}
            onDelete={onDelete}
            onSelect={onSelect}
            onEditLabels={this.handleStartEditingLabels}
          />
        ))}
      </>
    )
    return taskrows
  }

  private get sortedRows(): JSX.Element {
    const {tasks} = this.props
    const {sortKey, sortDirection} = this.state

    if (tasks.length) {
      return (
        <SortingHat<Task>
          list={tasks}
          sortKey={sortKey}
          direction={sortDirection}
        >
          {this.rows}
        </SortingHat>
      )
    }

    return null
  }

  private handleStartEditingLabels = (taskLabelsEdit: Task): void => {
    this.setState({taskLabelsEdit, isEditingTaskLabels: true})
  }

  private handleStopEditingLabels = (): void => {
    this.setState({isEditingTaskLabels: false})
  }

  private get renderLabelEditorOverlay(): JSX.Element {
    const {onAddTaskLabels, onRemoveTaskLabels} = this.props
    const {isEditingTaskLabels, taskLabelsEdit} = this.state

    return (
      <OverlayTechnology visible={isEditingTaskLabels}>
        <EditLabelsOverlay<Task>
          resource={taskLabelsEdit}
          onDismissOverlay={this.handleStopEditingLabels}
          onAddLabels={onAddTaskLabels}
          onRemoveLabels={onRemoveTaskLabels}
        />
      </OverlayTechnology>
    )
  }
}
