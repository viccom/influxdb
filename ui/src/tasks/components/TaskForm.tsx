// Libraries
import _ from 'lodash'
import React, {PureComponent, ChangeEvent} from 'react'

// Components
import {
  ComponentSpacer,
  Form,
  Grid,
  Columns,
  Input,
  Radio,
  ButtonShape,
  Button,
  ComponentColor,
  ButtonType,
} from 'src/clockface'
import TaskOptionsOrgDropdown from 'src/tasks/components/TasksOptionsOrgDropdown'
import TaskScheduleFormField from 'src/tasks/components/TaskScheduleFormField'

// Types
import {TaskOptions, TaskSchedule} from 'src/utils/taskOptionsToFluxScript'
import {Alignment, Stack, ComponentStatus} from 'src/clockface/types'
import {Organization} from 'src/api'

// Styles
import './TaskForm.scss'

interface Props {
  orgs: Organization[]
  taskOptions: TaskOptions
  onChangeScheduleType: (schedule: TaskSchedule) => void
  onChangeInput: (e: ChangeEvent<HTMLInputElement>) => void
  onChangeTaskOrgID: (orgID: string) => void
  isInOverlay?: boolean
  onSubmit?: () => void
  canSubmit?: boolean
  dismiss?: () => void
}

interface State {
  schedule: TaskSchedule
}

export default class TaskForm extends PureComponent<Props, State> {
  public static defaultProps: Partial<Props> = {
    isInOverlay: false,
    onSubmit: () => {},
    canSubmit: true,
    dismiss: () => {},
  }
  constructor(props) {
    super(props)

    this.state = {
      schedule: props.taskOptions.taskScheduleType,
    }
  }

  public render() {
    const {
      onChangeInput,
      onChangeTaskOrgID,
      taskOptions: {name, taskScheduleType, interval, offset, cron, orgID},
      orgs,
      isInOverlay,
    } = this.props

    return (
      <Form>
        <Grid>
          <Grid.Row>
            <Grid.Column widthXS={Columns.Twelve}>
              <Form.Element label="Name">
                <Input
                  name="name"
                  placeholder="Name your task"
                  onChange={onChangeInput}
                  value={name}
                />
              </Form.Element>
            </Grid.Column>
            <Grid.Column widthXS={Columns.Twelve}>
              <Form.Element label="Owner">
                <TaskOptionsOrgDropdown
                  orgs={orgs}
                  selectedOrgID={orgID}
                  onChangeTaskOrgID={onChangeTaskOrgID}
                />
              </Form.Element>
            </Grid.Column>
            <Grid.Column>
              <Form.Element label="Schedule Task">
                <ComponentSpacer
                  align={Alignment.Left}
                  stackChildren={Stack.Rows}
                >
                  <Radio shape={ButtonShape.StretchToFit}>
                    <Radio.Button
                      id="interval"
                      active={taskScheduleType === TaskSchedule.interval}
                      value={TaskSchedule.interval}
                      titleText="Interval"
                      onClick={this.handleChangeScheduleType}
                    >
                      Interval
                    </Radio.Button>
                    <Radio.Button
                      id="cron"
                      active={taskScheduleType === TaskSchedule.cron}
                      value={TaskSchedule.cron}
                      titleText="Cron"
                      onClick={this.handleChangeScheduleType}
                    >
                      Cron
                    </Radio.Button>
                  </Radio>
                </ComponentSpacer>
              </Form.Element>
            </Grid.Column>
            <TaskScheduleFormField
              onChangeInput={onChangeInput}
              schedule={taskScheduleType}
              interval={interval}
              offset={offset}
              cron={cron}
            />
            {isInOverlay && this.buttons}
          </Grid.Row>
        </Grid>
      </Form>
    )
  }

  private get buttons(): JSX.Element {
    const {onSubmit, canSubmit, dismiss} = this.props
    return (
      <Grid.Column widthXS={Columns.Twelve}>
        <Form.Footer>
          <Button
            text="Cancel"
            onClick={dismiss}
            titleText="Cancel save"
            type={ButtonType.Button}
          />
          <Button
            text={'Save as Task'}
            color={ComponentColor.Success}
            type={ButtonType.Submit}
            onClick={onSubmit}
            status={
              canSubmit ? ComponentStatus.Default : ComponentStatus.Disabled
            }
          />
        </Form.Footer>
      </Grid.Column>
    )
  }

  private handleChangeScheduleType = (schedule: TaskSchedule): void => {
    this.props.onChangeScheduleType(schedule)
  }
}
