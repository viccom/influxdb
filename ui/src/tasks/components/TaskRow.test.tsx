// Libraries
import React from 'react'
import {shallow} from 'enzyme'

// Components
import {TaskRow} from 'src/tasks/components/TaskRow'

// Types
import {Label} from 'src/clockface'

// Constants
import {tasks, withRouterProps} from 'mocks/dummyData'

const setup = (override = {}) => {
  const props = {
    ...withRouterProps,
    task: tasks[0],
    onActivate: jest.fn(),
    onDelete: jest.fn(),
    onSelect: jest.fn(),
    onEditLabels: jest.fn(),
    ...override,
  }

  const wrapper = shallow(<TaskRow {...props} />)

  return {wrapper}
}

describe('Tasks.Components.TaskRow', () => {
  it('renders', () => {
    const {wrapper} = setup()
    expect(wrapper.exists()).toBe(true)
    expect(wrapper).toMatchSnapshot()
  })

  describe('if task has labels', () => {
    it('renders with labels', () => {
      const {wrapper} = setup({task: tasks[1]})

      const labelContainer = wrapper.find(Label.Container)
      const labels = wrapper.find(Label)

      expect(labelContainer.exists()).toBe(true)
      expect(labels.length).toBe(tasks[1].labels.length)
    })
  })
})
