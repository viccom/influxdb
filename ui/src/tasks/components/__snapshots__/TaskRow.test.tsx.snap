// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tasks.Components.TaskRow renders 1`] = `
<IndexListRow
  disabled={false}
>
  <IndexListRowCell
    alignment="left"
    revealOnHover={false}
  >
    <ComponentSpacer
      align="left"
      stackChildren="rows"
    >
      <a
        href="#"
        onClick={[Function]}
      >
        pasdlak
      </a>
      <LabelContainer
        className="index-list--labels"
        limitChildCount={4}
        onEdit={[Function]}
        resourceName="this resource"
      />
    </ComponentSpacer>
  </IndexListRowCell>
  <IndexListRowCell
    alignment="left"
    revealOnHover={false}
  >
    <SlideToggle
      active={true}
      color="primary"
      disabled={false}
      onChange={[Function]}
      size="xs"
      tooltipText=""
    />
  </IndexListRowCell>
  <IndexListRowCell
    alignment="left"
    revealOnHover={false}
  >
    2 0 * * *
  </IndexListRowCell>
  <IndexListRowCell
    alignment="left"
    revealOnHover={false}
  >
    <a
      href=""
      onClick={[Function]}
    >
      RadicalOrganization
    </a>
  </IndexListRowCell>
  <IndexListRowCell
    alignment="right"
    revealOnHover={true}
  >
    <ComponentSpacer
      align="right"
    >
      <Button
        active={false}
        color="default"
        icon="export"
        onClick={[Function]}
        shape="none"
        size="xs"
        status="default"
        text="Export"
        type="button"
      />
      <ConfirmationButton
        confirmText="Confirm"
        onConfirm={[MockFunction]}
        returnValue={
          Object {
            "cron": "2 0 * * *",
            "flux": "option task = {
  name: \\"pasdlak\\",
  cron: \\"2 0 * * *\\"
}
from(bucket: \\"inbucket\\") 
|> range(start: -1h)",
            "id": "02ef9deff2141000",
            "labels": Array [],
            "name": "pasdlak",
            "orgID": "02ee9e2a29d73000",
            "organization": Object {
              "id": "02ee9e2a29d73000",
              "links": Object {
                "buckets": "/api/v2/buckets?org=RadicalOrganization",
                "dashboards": "/api/v2/dashboards?org=RadicalOrganization",
                "self": "/api/v2/orgs/02ee9e2a29d73000",
                "tasks": "/api/v2/tasks?org=RadicalOrganization",
              },
              "name": "RadicalOrganization",
            },
            "owner": Object {
              "id": "02ee9e2a19d73000",
              "name": "",
            },
            "status": "active",
          }
        }
        shape="none"
        size="xs"
        status="default"
        text="Delete"
      />
    </ComponentSpacer>
  </IndexListRowCell>
</IndexListRow>
`;
